pepper-apis/src/test/java/org/broadinstitute/ddp/RateLimitTest.java

```diff
@@ -0,0 +1,127 @@
+        }
+    }
+
+    /**
+     * Before each test, we shut down and reboot the backend
+     * so that we clear DOSFilter's current rate calculations.
+     */
+    @Before
+    public void shutdownDDPAndLowerRateLimits() {
+        tearDown();
+
+        ConfigManager configManager = ConfigManager.getInstance();
+
+        configManager.overrideValue(ConfigFile.API_RATE_LIMIT.MAX_QUERIES_PER_SECOND, "1");
+        configManager.overrideValue(ConfigFile.API_RATE_LIMIT.BURST, "1");
+
+        startupTestServer();
+
+        url = RouteTestUtil.getTestingBaseUrl() + RouteConstants.API.HEALTH_CHECK;
+        password = RouteTestUtil.getConfig().getString(ConfigFile.HEALTHCHECK_PASSWORD);
+    }
+
+    @Test
+    public void testThingsUnderTheLimitStillGetThrough() {
+        runHealthChecksConcurrentlyAndCheckResults(1, 0);
+    }
+
+    @Test
+    public void testThingsOverTheLimitAreRejected() {
+        runHealthChecksConcurrentlyAndCheckResults(10, 2);
+    }
+
+    @AfterClass
+    public static void resetConfig() {
+        tearDown();
+        for (Map.Entry<String, String> originalConfig : originalRateLimitConfig.entrySet()) {
+            LOG.info("Resetting rate limit {} to {}", originalConfig.getKey(), originalConfig.getValue());
+            ConfigManager.getInstance().overrideValue(originalConfig.getKey(), originalConfig.getValue());
+        }
+        startupTestServer();
+    }
+
+    private static void runHealthChecksConcurrentlyAndCheckResults(int numConcurrentQueries, int minimumFailures) {
+        ExecutorService executorService = Executors.newFixedThreadPool(numConcurrentQueries);
+        AtomicInteger rateLimitRejects = new AtomicInteger(0);
+
+        List<Callable<Boolean>> healthCheckCalls = new ArrayList<>();
+
+        for (int i = 0; i < numConcurrentQueries; i++) {
+            healthCheckCalls.add(() -> {
+               HttpResponse httpResponse = null;
+               try {
+                   httpResponse = Request.Get(url).addHeader("Host", password).execute().returnResponse();
+                   LOG.info(httpResponse.getStatusLine().getReasonPhrase());
+               } catch (IOException e) {
+                   LOG.info("Rate limit query test exception", e);
+               }
+               if (httpResponse != null && httpResponse.getStatusLine().getStatusCode() == HttpStatus.TOO_MANY_REQUESTS_429) {
+                   rateLimitRejects.incrementAndGet();
+                   return false;
+               }
+               return true;
+            });
+        }
+
+        try {
+            executorService.invokeAll(healthCheckCalls);
+        } catch (InterruptedException e) {
+            LOG.error("Interrupted while running rate limit filter test", e);
+        }
+
+        executorService.shutdown();
+
+        Assert.assertTrue("Expected at least " + minimumFailures + " but got " + rateLimitRejects.get(),
+                rateLimitRejects.get() >= minimumFailures);
+    }
+}
```

Comment:
This is a really nice test!
2 small requests:
1. Check the the response body: there should be a health check response if succeeds, and there should not be one when it fails. Want to make sure that we are really not getting to the route when rejecting.
2. Since this is an integration test, could you verify that at least one of the other filters are still working (I would check that the AuthFilter is still operational. Just do an unauthenticated request and check for that 401 code)

Reply:
Got it.  Will address both.

Reply:
Changing this a bit more now...

Reply:
@MocanaAtBroad I think I've covered the bases here now.  Body is checked, and there's some checking of the internal state of a different filter to verify it is hit (or not) in different situations.